### PR TITLE
Vorpal in VsCode is unable to run scans in Sample with complex names (AST-62777)

### PR DESCRIPTION
**Pull request by elchananarb**

### CODE REVIEW

**Comment by AlvoBen** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [d1ef1c0e\-30b3\-4ed2\-a44b\-125d3e2f2581](https://eu.ast.checkmarx.net/projects/800f0cfd-da0f-47f5-b9db-feb25b4bc73a/scans?id=d1ef1c0e-30b3-4ed2-a44b-125d3e2f2581&branch=bug%2Felchanan%2Fscan-failure-complex-filenames)
### No New Or Fixed Issues Found

File: src/main/wrapper/ExecutionService.ts [hunk 1]
```diff
@@ -35,12 +36,24 @@ function isJsonString(s: string) {
...
 }
 
 function transform(n:string) {
+    // If the flag is set, return the current string as-is and reset the flag
+    if (skipNext) {
+        skipNext = false;
+        return n;
+    }
+    // If the current string is "--file-source", set the flag and return it as-is
+    if (n === "--file-source") {
```
Comment (OrShamirCM) [inline src/main/wrapper/ExecutionService.ts:52]: can you use a common const for - "--file-source" ?

File: src/main/wrapper/ExecutionService.ts [hunk 2]
```diff
@@ -35,12 +36,24 @@ function isJsonString(s: string) {
     return true;
 }
 
-function transformation(commands: string[]):string[] {
-    const result:string[] = commands.map(transform);
+function transformation(commands: string[]): string[] {
+    skipNext = false; // Reset the flag before processing
```
Comment (OrShamirCM) [inline src/main/wrapper/ExecutionService.ts:40]: skipValue maybe?

Comment (elchananarb) [reply src/main/wrapper/ExecutionService.ts:40]: Sounds good

File: src/main/wrapper/ExecutionService.ts (hunk 1 above)
Comment (elchananarb) [reply src/main/wrapper/ExecutionService.ts:52]: done

File: src/main/wrapper/ExecutionService.ts [hunk 3]
```diff
@@ -24,6 +24,8 @@ import CxChat from "../chat/CxChat";
 import CxMask from "../mask/CxMask";
 import CxVorpal from "../vorpal/CxVorpal";
 
+let skipValue = false;
```
Comment (AlvoBen) [inline src/main/wrapper/ExecutionService.ts:27]: Can you make skipValue variable of transofrmation function and pass it as a parameter to transform? I think it's better practice then global variable

**Review by AlvoBen — APPROVED** [review]

File: src/tests/ScanTest.test.ts [hunk 4]
```diff
@@ -145,6 +145,16 @@ describe("ScanCreate cases", () => {
         expect(scanObject.status).toEqual(true);
     });
 
+    it('ScanVorpal with complex name Successful case', async () => {
```
Comment (AlvoBen) [inline src/tests/ScanTest.test.ts:148]: Nice test 👑

File: src/main/wrapper/ExecutionService.ts (hunk 3 above)
Comment (elchananarb) [reply src/main/wrapper/ExecutionService.ts:27]: We will insert it later so as not to delay the release

**Pull request merged**